File: 0228-summary-ranges/0228-summary-ranges.cpp
class Solution {
public:
    vector<string> summaryRanges(vector<int>& nums) {
        vector<string> list;

        int i = 0;
        while(i < nums.size()){
            int start = nums[i];
            while((i < nums.size()-1) && (long long)nums[i+1] - nums[i] == 1){
                i++;
            }

            int end = nums[i];

            if(start == end){
                list.push_back(to_string(start));
            }else{
                list.push_back(to_string(start) + "->" + to_string(end));
            }

            i++;
        }

        return list;
    }
};

File: 0228-summary-ranges/0228-summary-ranges.java
class Solution {
    public List<String> summaryRanges(int[] nums) {
        List<String> str = new ArrayList<>();

        int i = 0;
        while (i < (nums.length)) {
            int start = nums[i];
            while (i < (nums.length-1) && (nums[i + 1] - nums[i]) == 1) {
                i++;
            }

            int end = nums[i];

            if(start == end){
                str.add(String.valueOf(start));
            }else{
                str.add(start +"->"+ end);
            }

            i++;
        }
        return str;
    }
}